helpers/src/twap/tests/unit.rs

```diff
@@ -0,0 +1,499 @@
+#[case(9 * HOUR, 6)]
+#[case(10 * HOUR, 6)]
+#[case(11 * HOUR, 6)]
+#[case(12 * HOUR, 6)]
+#[case(1 * DAY, 6)]
+#[case(2 * DAY, 6)]
+#[case(3 * DAY, 6)]
+#[case(4 * DAY, 6)]
+#[case(5 * DAY, 6)]
+#[case(6 * DAY, 6)]
+#[case(7 * DAY, 6)]
+#[case(1 * HOUR, 7)]
+#[case(2 * HOUR, 7)]
+#[case(3 * HOUR, 7)]
+#[case(4 * HOUR, 7)]
+#[case(5 * HOUR, 7)]
+#[case(6 * HOUR, 7)]
+#[case(7 * HOUR, 7)]
+#[case(8 * HOUR, 7)]
+#[case(9 * HOUR, 7)]
+#[case(10 * HOUR, 7)]
+#[case(11 * HOUR, 7)]
+#[case(12 * HOUR, 7)]
+#[case(1 * DAY, 7)]
+#[case(2 * DAY, 7)]
+#[case(3 * DAY, 7)]
+#[case(4 * DAY, 7)]
+#[case(5 * DAY, 7)]
+#[case(6 * DAY, 7)]
+#[case(7 * DAY, 7)]
+#[case(1 * HOUR, 8)]
+#[case(2 * HOUR, 8)]
+#[case(3 * HOUR, 8)]
+#[case(4 * HOUR, 8)]
+#[case(5 * HOUR, 8)]
+#[case(6 * HOUR, 8)]
+#[case(7 * HOUR, 8)]
+#[case(8 * HOUR, 8)]
+#[case(9 * HOUR, 8)]
+#[case(10 * HOUR, 8)]
+#[case(11 * HOUR, 8)]
+#[case(12 * HOUR, 8)]
+#[case(1 * DAY, 8)]
+#[case(2 * DAY, 8)]
+#[case(3 * DAY, 8)]
+#[case(4 * DAY, 8)]
+#[case(5 * DAY, 8)]
+#[case(6 * DAY, 8)]
+#[case(7 * DAY, 8)]
+#[case(1 * HOUR, 9)]
+#[case(2 * HOUR, 9)]
+#[case(3 * HOUR, 9)]
+#[case(4 * HOUR, 9)]
+#[case(5 * HOUR, 9)]
+#[case(6 * HOUR, 9)]
+#[case(7 * HOUR, 9)]
+#[case(8 * HOUR, 9)]
+#[case(9 * HOUR, 9)]
+#[case(10 * HOUR, 9)]
+#[case(11 * HOUR, 9)]
+#[case(12 * HOUR, 9)]
+#[case(1 * DAY, 9)]
+#[case(2 * DAY, 9)]
+#[case(3 * DAY, 9)]
+#[case(4 * DAY, 9)]
+#[case(5 * DAY, 9)]
+#[case(6 * DAY, 9)]
+#[case(7 * DAY, 9)]
+#[case(1 * HOUR, 10)]
+#[case(2 * HOUR, 10)]
+#[case(3 * HOUR, 10)]
+#[case(4 * HOUR, 10)]
+#[case(5 * HOUR, 10)]
+#[case(6 * HOUR, 10)]
+#[case(7 * HOUR, 10)]
+#[case(8 * HOUR, 10)]
+#[case(9 * HOUR, 10)]
+#[case(10 * HOUR, 10)]
+#[case(11 * HOUR, 10)]
+#[case(12 * HOUR, 10)]
+#[case(1 * DAY, 10)]
+#[case(2 * DAY, 10)]
+#[case(3 * DAY, 10)]
+#[case(4 * DAY, 10)]
+#[case(5 * DAY, 10)]
+#[case(6 * DAY, 10)]
+#[case(7 * DAY, 10)]
+#[case(1 * HOUR, 11)]
+#[case(2 * HOUR, 11)]
+#[case(3 * HOUR, 11)]
+#[case(4 * HOUR, 11)]
+#[case(5 * HOUR, 11)]
+#[case(6 * HOUR, 11)]
+#[case(7 * HOUR, 11)]
+#[case(8 * HOUR, 11)]
+#[case(9 * HOUR, 11)]
+#[case(10 * HOUR, 11)]
+#[case(11 * HOUR, 11)]
+#[case(12 * HOUR, 11)]
+#[case(1 * DAY, 11)]
+#[case(2 * DAY, 11)]
+#[case(3 * DAY, 11)]
+#[case(4 * DAY, 11)]
+#[case(5 * DAY, 11)]
+#[case(6 * DAY, 11)]
+#[case(7 * DAY, 11)]
+#[case(1 * HOUR, 12)]
+#[case(2 * HOUR, 12)]
+#[case(3 * HOUR, 12)]
+#[case(4 * HOUR, 12)]
+#[case(5 * HOUR, 12)]
+#[case(6 * HOUR, 12)]
+#[case(7 * HOUR, 12)]
+#[case(8 * HOUR, 12)]
+#[case(9 * HOUR, 12)]
+#[case(10 * HOUR, 12)]
+#[case(11 * HOUR, 12)]
+#[case(12 * HOUR, 12)]
+#[case(1 * DAY, 12)]
+#[case(2 * DAY, 12)]
+#[case(3 * DAY, 12)]
+#[case(4 * DAY, 12)]
+#[case(5 * DAY, 12)]
+#[case(6 * DAY, 12)]
+#[case(7 * DAY, 12)]
+#[ignore = "TODO: Still need to configure flag to run this test as this \
+            one takes quite a while and is not meant to be run every time"]
+fn test_polars(#[case] period: u64, #[case] dataset: i32) {
```

Comment:
Can't give too much feedback on this, but what are you testing for in general terms?

Reply:
Yeah, it makes sense having a separate repo in the organization for data analysis, I agree